src/main/java/com/google/devtools/build/lib/rules/cpp/CcLibrary.java

```diff
@@ -620,13 +630,20 @@ public static boolean appearsToHaveObjectFiles(AttributeMap rule) {
 
     if (!ruleContext.attributes().get("linkstatic", Type.BOOLEAN)
         && !ccCompilationOutputs.isEmpty()) {
+      String linkedArtifactNameSuffix = "";
+      if(cppConfiguration.isRenameDLL()) {
```

Comment:
The code for calculating the dll suffix repeated twice in the change. Can you extract it to a function? Maybe in CppHelper?